app/uk/gov/hmrc/gform/models/FormModelBuilder.scala

```diff
@@ -278,7 +278,8 @@ class FormModelBuilder[E, F[_]: Functor](
     }
 
     val visibleTypedExprs: List[(FormComponentId, TypeInfo)] = visibilityFormModel.allFormComponents.collect {
-      case fc @ HasValueExpr(expr) if !fc.editable => (fc.id, visibilityFormModel.explicitTypedExpr(expr, fc.id))
+      case fc @ HasValueExpr(expr) if !fc.editable =>
+        (fc.id, visibilityFormModel.toFirstOperandTypeInfo(expr))
```

Comment:
I'm afraid this is wrong. `TypeInfo` for component with value expression needs to be derived from component, not from the expression.

For example type of expression `${1 + 2}` is `String`, but type of expression `${4 + 5}` is `Number`:

```json
{
    "id": "foo",
    "type":"text",
    "format": "shortText",
    "value": "${1 + 2}",
    "label": "Foo ${4 + 5}"
}
```

Method `toFirstOperandTypeInfo` is supposed to derive expression type from free expressions, those are expression in labels etc.